lettermgmt/models/res_letter.py

```diff
@@ -1,165 +1,200 @@
+        string='Expeditor Reference',
+        help="Reference Number used by Expeditor.")
+
+    parent_id = fields.Many2one(
+        'res.letter',
+        string='Parent',
+        groups='lettermgmt.group_letter_thread')
+    child_line = fields.One2many(
+        'res.letter',
+        'parent_id',
+        string='Letter Lines',
+        groups='lettermgmt.group_letter_thread')
+
+    reassignment_ids = fields.One2many(
+        'letter.reassignment',
+        'letter_id',
+        string='Reassignment lines',
+        help='Reassignment users and comments',
+        groups='lettermgmt.group_letter_reasignment')
+
+    # This field seems to be unused. TODO: Remove it?
+    extern_partner_ids = fields.Many2many(
+        'res.partner',
+        string='Recipients')
+
+    @api.model
+    def create(self, vals):
+        if ('number' not in vals) or (vals.get('number') in ('/', False)):
+            sequence = self.env['ir.sequence']
+            move_type = vals.get('move', self.env.context.get(
+                'default_move', self.env.context.get('move', 'in')))
+            vals['number'] = sequence.get('%s.letter' % move_type)
+        return super(ResLetter, self).create(vals)
 
-    def action_create(self, cr, uid, ids, context=None):
-        """Put the state of the letter into Crated"""
-        for letter in self.browse(cr, uid, ids, context=context):
-            self.write(
-                cr, uid, [letter.id], {'state': 'created'}, context=context)
+    @api.multi
+    def action_cancel(self):
+        """ Put the state of the letter into Cancelled """
+        self.write({'state': 'cancel'})
         return True
 
-    def action_validate(self, cr, uid, ids, context=None):
-        """Put the state of the letter into Validated"""
-        for letter in self.browse(cr, uid, ids, context=context):
-            self.write(
-                cr, uid, [letter.id], {'state': 'validated'}, context=context)
+    @api.multi
+    def action_cancel_draft(self):
+        """ Go from cancelled state to draf state """
+        self.write({'state': 'draft'})
+        self.delete_workflow()
+        self.create_workflow()
         return True
 
-    def action_send(self, cr, uid, ids, context=None):
-        """Put the state of the letter into sent"""
-        for letter in self.browse(cr, uid, ids, context=context):
-            self.write(
-                cr, uid, [letter.id],
-                {
-                    'state': 'sent',
-                    'snd_rec_date': letter.snd_rec_date or
-                    fields.datetime.now()
-                },
-                context=context)
-            if letter.sender_partner_id:
-                letter.message_subscribe([letter.sender_partner_id.id])
+    @api.multi
+    def action_send(self):
+        """ Put the state of the letter into sent """
+        self.write({
+            'state': 'sent',
+            'snd_date': self.snd_date or fields.Date.today()
```

Comment:
Yeah.. sorry about that. I should've reviewed past me myself.
I'll fix all of this later this week, probably Friday.